tensorflow_datasets/scripts/replace_fake_images.py

```diff
@@ -116,8 +115,8 @@ def rewrite_tar(root_dir, tar_filepath):
     rewrite_dir(temp_dir)  # Recursivelly compress the archive content
 
     # Converting into tarfile again to decrease the space taken by the file-
-    with tarfile.open(tar_filepath, 'w' + extension) as tar:
-      tar.add(temp_dir, recursive=True)
+    with tarfile.open(tar_filepath, 'w:gz') as tar:
```

Comment:
I don't think we should compress all tar files using gz. `.tar.gz` are compressed with `.gz`, but not `.tar` files.

Reply:
Actually compressing files with `.gz` compress files better, if we don't use it here it ends up in large files as previous. But I provided `arcname` so that final output file is of format `.tar`.

I tried out many things but it works finally is it okay to use this method ?

Reply:
@Conchylicultor  is it ok if I compress files using .gz and saving it as .tar final file was saving as .tar not .gz

Reply:
I think this may creates bug when the program expect tar while it would actually receive targz compressed files.

Reply:
Is there a way to detect if the original `.tar` archive file was compressed with `gz` or not ? If so, we should apply the same compression as the original archive.

Reply:
@Conchylicultor  Yeah sure I fill look that, but I after compressing files with .gz I tried to decompress it using `tar -xzvf archive.tar` and it gives decompresses them successfully as original one, Is it make any sense ?